### PR TITLE
DEV - Simplify Docker images

### PR DESCRIPTION
**Pull request by trallard**
This is a stack on top of #838 

## Description

This PR continues to simplify the development and test setup for conda-store

- Removes unnecessary `environment.yaml` files (less files to keep in check/updated)
- Simplifies and adds comments to Dockerfiles
- Bumps the base image to `[24.3.0-0](https://hub.docker.com/layers/condaforge/mambaforge/24.3.0-0/images/sha256-21f7908a6c984ae164cb51c8af71ef9901693630f4b744038e5df6ef968d9b05?context=explore)`

## Pull request checklist

- [x] Did you test this change locally?
- [ ] Did you update the documentation (if required)?
- [ ] Did you add/update relevant tests for this change (if required)?

## Additional information

Since this stacks on another PR, some integration and unit tests might fail (and will be fixed in #838). For now, the important parts are :
* Docker-related workflows, which are working.
* Integration tests (conda-store and conda-store-server) - in the sense that the container should be building and `docker compose up` should be working (which it is)

I built the Docker images locally, then used these ones for `docker-compose`, and all seemed to work.

### CODE REVIEW

File: conda-store-server/Dockerfile [hunk 1]
```diff
@@ -39,28 +41,41 @@ RUN printf 'channels: []\n' > /opt/conda/.condarc && \
...
-COPY environment.yaml /opt/conda-store-server/environment.yaml
+RUN mamba create --name ${conda_env_name} \
+    python=${python_version} conda \
+    --channel conda-forge -y && \
+    conda clean --force-pkgs-dirs --all -y
 
-RUN mamba env create -f /opt/conda-store-server/environment.yaml && \
-	conda clean  --force-pkgs-dirs
+# ensure we are using the conda environment
+ENV PATH=/opt/conda/condabin:/opt/conda/envs/conda-store-server/bin:/opt/conda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:${PATH}
```
Comment (peytondmurray) [inline conda-store-server/Dockerfile:50]: Hmm, why do we need this now? Was `PATH` not correctly set in the previous dockerfile?

File: conda-store-server/Dockerfile [hunk 2]
```diff
@@ -1,10 +1,12 @@
-FROM condaforge/mambaforge:23.3.1-1 as base
+FROM condaforge/mambaforge:24.3.0-0 as base
 
 LABEL org.opencontainers.image.authors="conda-store development team"
 
-ENV PATH=/opt/conda/condabin:/opt/conda/envs/conda-store-server/bin:/opt/conda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:${PATH}
 ENV TZ=Etc/UTC
 
+ARG python_version=3.10
```
[the remark this answers is not in the document]
Comment (trallard) [reply conda-store-server/Dockerfile:7]: We have 3.12 as the default everywhere (dev environment, CI, etc.) I am happy moving this to a newer version but would prefer to keep all of them in sync

File: conda-store-server/Dockerfile (hunk 1 above)
Comment (trallard) [reply conda-store-server/Dockerfile:50]: I just moved its placement so it was the same for both Dockerfiles; I can move these both to the top if that is more intuitive

Comment (trallard) [reply conda-store-server/Dockerfile]: nvm, moved it so we have all the envs together at the top

File: .dockerignore [hunk 3]
```diff
@@ -1,2 +1,6 @@
 data
 examples/docker/data
+
+# ignore test files
+conda-store-server/tests
+conda-store/tests
```
[the remark this answers is not in the document]
Comment (trallard) [reply .dockerignore]: And found even more that should be ignored.

**Review by peytondmurray — APPROVED** [review]
Thanks for clearing up the questions I had :ship:

**Pull request merged**